packages/bar/src/BarTotals.tsx

```diff
@@ -0,0 +1,154 @@
+import { AnyScale, ScaleBand } from '@nivo/scales'
+import { defaultProps } from './props'
+import { BarCommonProps, BarDatum, ComputedBarDatum } from './types'
+
+interface BarTotalsProps<RawDatum extends BarDatum> {
+    bars: ComputedBarDatum<RawDatum>[]
+    xScale: ScaleBand<string> | AnyScale
+    yScale: ScaleBand<string> | AnyScale
+    layout?: BarCommonProps<RawDatum>['layout']
+    groupMode?: BarCommonProps<RawDatum>['groupMode']
+}
+
+export const BarTotals = <RawDatum extends BarDatum>({
+    bars,
+    xScale,
+    yScale,
+    layout = defaultProps.layout,
+    groupMode = defaultProps.groupMode,
+}: BarTotalsProps<RawDatum>) => {
+    if (bars.length === 0) return <></>
+    const totals: JSX.Element[] = []
+
+    const totalsByIndex = new Map<string | number, number>()
+
+    const barWidth = bars[0].width
+    const barHeight = bars[0].height
+    const yOffsetVertically = -10
```

Comment:
As for the canvas implementation, styles should come from the theme and offsets should be configurable.

Reply:
Done! Thanks for the tip